src/openms/source/FILTERING/DATAREDUCTION/Deisotoper.cpp

```diff
@@ -113,7 +118,7 @@ void Deisotoper::deisotopeAndSingleCharge(MSSpectrum& spec,
           {
             // Possible improvement: include proper averagine model filtering. for now start at the second peak to test hypothesis
             // Note: this is a common approach used in several other search engines
-            if (old_spectrum[p].getIntensity() > old_spectrum[extensions.back()].getIntensity())
+            if (model_decreasing && old_spectrum[p].getIntensity() > old_spectrum[extensions.back()].getIntensity())
```

Comment:
@timosachsenberg how would this work if the second peak is the highest one? Wouldnt it just abort?

Reply:
hmm, maybe @cbielow changed that?

Reply:
this would clearly be a regression (actually it states in the comment "start at the second peak to test hypothesis")

Reply:
I think it should read:
if (n_extensions >=2)

Reply:
Yes I see but I think this is already present in a very early version: 016a63d62ed9ec5440d6ec8ab2897344efc11ba1

Also I think @enetz had similar issues in his code and commented out this part: https://github.com/OpenMS/OpenMS/blob/5c874d295aaa958f67b81f54ee101ce48b4c9394/src/openms/source/ANALYSIS/XLMS/OPXLSpectrumProcessingAlgorithms.cpp#L565-L578

maybe it would make more sense to use the `Deisotoper` in his code and just disable this part of the code? Any comment @enetz

Reply:
@hroest I have been tinkering with my version of the Deisotoper a bit, e.g. my function reports the number of isotopic peaks found for each pattern and adds up the intensities of all isotopic peaks into the monoisotopic peak. I can try to combine the deisotopers and maybe add two more flags for this additional functionality.

Reply:
> @hroest I have been tinkering with my version of the Deisotoper a bit, e.g. my function reports the number of isotopic peaks found for each pattern and adds up the intensities of all isotopic peaks into the monoisotopic peak. I can try to combine the deisotopers and maybe add two more flags for this additional functionality.

yes, this would be great as an option. Thanks!